server/src/main/java/org/elasticsearch/cluster/metadata/TemplateUpgradeService.java

```diff
@@ -151,6 +151,12 @@ void updateTemplates(Map<String, BytesReference> changes, Set<String> deletions)
             client.admin().indices().putTemplate(request, new ActionListener<PutIndexTemplateResponse>() {
                 @Override
                 public void onResponse(PutIndexTemplateResponse response) {
+                    // check successful template change
+                    final Optional<Tuple<Map<String, BytesReference>, Set<String>>> afterChanges =
+                            calculateTemplateChanges(clusterService.state().getMetaData().getTemplates());
```

Comment:
Does it really make sense to recalculate the changes after every update (/delete)?
Can't we just do it once when `updatesInProgress.decrementAndGet() == 0` ?

Reply:
Admittedly it is overkill to validate each template change.
The final check, however, has to take into account that some updates have not been acknowledged therefore template changes should still be reported.